### ISSUE TITLE
Checking for package NumpyPk when installing UW on the server

### ISSUE THREAD

**Issue by tingyang2004** [underworldcode/underworld2#365]
Hi all,

My school IT technician met this issue when installing UW2 on the school server, any suggestions? Let me know if you need more information. Thanks a lot.

Best regards,
Ting
```bash
[root@mn01 libUnderworld]# ./configure.py --with-glucifer=0 --with-debugging=0 --python-dir=/share/apps/python/2.7.15 --hdf5-dir=/share/base/hdf5/1.10.5-gcc-4.8.5-mpich-3.3 --petsc-dir=/share/base/petsc/3.10.5-gcc-4.8.5-mpich-3.3 --mpi-dir=/share/base/mpich/3.3-gcc-4.8.5 --prefix=/share/tools/underworld2
scons: Reading SConscript files ...
Checking whether the C compiler worksyes
Checking for Solvers... yes
Checking for package libm... yes
Checking for package libXML2... yes
Checking for package MPI... yes
Exception RuntimeError: 'generator ignored GeneratorExit' in <generator object gen_envs at 0x2ba33550aaf0> ignored
Checking for package HDF5... yes
Checking for package Python... yes
Checking for package PETSc... yes
IndexError: list index out of range:
  File "/share/home/tche/software/underworld2/libUnderworld/SConfigure", line 178:
    env.ConfigurePackage(packages.NumpyPk, required=True)
  File "/share/home/tche/software/underworld2/libUnderworld/config/scons/scons-local-1.2.0/SCons/Environment.py", line 217:
    return apply(self.method, nargs, kwargs)
  File "/share/home/tche/software/underworld2/libUnderworld/config/config.py", line 29:
    pkg(**kw)
  File "/share/home/tche/software/underworld2/libUnderworld/config/Package.py", line 148:
    for env in self.gen_envs(loc):
  File "/share/home/tche/software/underworld2/libUnderworld/config/packages/NumpyPk.py", line 32:
    env.AppendUnique(CPPPATH=[self.location[1][0]])
Checking for package NumpyPk... 
```

**Comment by jmansour** [comment]
Hey @tingyang2004.  Ok, I'm not 100% sure, but i'll note that the `--prefix` option is currently not supported and the code needs to be built _in-place_.  There's a fair chance this is the cause of the issue you are encountering. 

I'll start another ticket to either deprecate or fix `--prefix`.

Let us know if removing this option gets you further.

**Comment by tingyang2004** [comment]
Thanks John, will give it a try and let you know the result.

**Comment by tingyang2004** [comment]
The school HPC technician has just tried to remove prefix option away, and the error message persists.

Yes, I have a try to remove the option --prefix, but the same error as before remains:

[root@mn01 libUnderworld]# echo $PATH
/share/base/petsc/3.10.5-gcc-4.8.5-mpich-3.3/lib/pestc/bin:/share/base/hdf5/1.10.5-gcc-4.8.5-mpich-3.3/bin:/share/base/mpich/3.3-gcc-4.8.5/bin:/share/apps/python/2.7.15/bin:/opt/xcat/bin:/opt/xcat/sbin:/opt/xcat/share/xcat/tools:/opt/ibm/lsfsuite/lsf/10.1/linux2.6-glibc2.3-x86_64/etc:/opt/ibm/lsfsuite/lsf/10.1/linux2.6-glibc2.3-x86_64/bin:/opt/ibm/lsfsuite/ext/perf/1.2/bin:/opt/ibm/lsfsuite/ext/gui/3.0/wlp/bin:/opt/ibm/lsfsuite/ext/gui/3.0/bin:/opt/ibm/lsfsuite/ext/rule-engine/1.0/linux2.6-glibc2.5-x86_64/bin:/opt/ibm/lsfsuite/ext/ppm/10.2/bin:/opt/ibm/lsfsuite/ext/ppm/10.2/linux2.6-glibc2.3-x86_64/bin:/opt/lenovo/onecli:/usr/lpp/mmfs/bin:/share/base/bin:/opt/confluent/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/opt/ibutils/bin:/bin:/usr/bin:/opt/ibm/jre/bin:/root/bin:/opt/StorageManager/jre/bin
[root@mn01 libUnderworld]# echo $LD_LIBRARY_PATH 
/share/base/petsc/3.10.5-gcc-4.8.5-mpich-3.3/lib:/share/base/hdf5/1.10.5-gcc-4.8.5-mpich-3.3/lib:/share/base/mpich/3.3-gcc-4.8.5/lib:/share/apps/python/2.7.15/lib:/opt/ibm/lsfsuite/ext/perf/1.2/linux-x86_64/lib:/opt/ibm/lsfsuite/ext/perf/ego/1.2/linux-x86_64/lib:/opt/ibm/lsfsuite/ext/perf/lsf/10.0/linux-x86_64/lib:/opt/ibm/lsfsuite/ext/perf/license/7.0/linux-x86_64/lib:/opt/ibm/lsfsuite/lsf/1.2/linux2.6-glibc2.3-x86_64/lib:/opt/ibm/lsfsuite/ext/ppm/10.2/linux2.6-glibc2.3-x86_64/lib:/opt/ibm/lsfsuite/lsf/10.1/linux2.6-glibc2.3-x86_64/lib:/opt/ibm/lsfsuite/lsf/10.1/linux2.6-glibc2.3-x86_64/lib
[root@mn01 libUnderworld]# 
[root@mn01 libUnderworld]# pwd
/share/tools/underworld2/libUnderworld
[root@mn01 libUnderworld]# ./configure.py --with-glucifer=0 --with-debugging=0 --python-dir=/share/apps/python/2.7.15 --hdf5-dir=/share/base/hdf5/1.10.5-gcc-4.8.5-mpich-3.3 --petsc-dir=/share/base/petsc/3.10.5-gcc-4.8.5-mpich-3.3 --mpi-dir=/share/base/mpich/3.3-gcc-4.8.5
scons: Reading SConscript files ...
Checking whether the C compiler worksyes
Checking for Solvers... yes
Checking for package libm... yes
Checking for package libXML2... yes
Checking for package MPI... yes
Exception RuntimeError: 'generator ignored GeneratorExit' in <generator object gen_envs at 0x2b86bb061af0> ignored
Checking for package HDF5... yes
Checking for package Python... yes
Checking for package PETSc... yes
IndexError: list index out of range:
  File "/share/tools/underworld2/libUnderworld/SConfigure", line 178:
    env.ConfigurePackage(packages.NumpyPk, required=True)
  File "/share/tools/underworld2/libUnderworld/config/scons/scons-local-1.2.0/SCons/Environment.py", line 217:
    return apply(self.method, nargs, kwargs)
  File "/share/tools/underworld2/libUnderworld/config/config.py", line 29:
    pkg(**kw)
  File "/share/tools/underworld2/libUnderworld/config/Package.py", line 148:
    for env in self.gen_envs(loc):
  File "/share/tools/underworld2/libUnderworld/config/packages/NumpyPk.py", line 32:
    env.AppendUnique(CPPPATH=[self.location[1][0]])
Checking for package NumpyPk...

**Comment by jmansour** [comment]
Can you get your admin to forward the `config.log` to you, and post it here?

**Comment by tingyang2004** [comment]
The config.log file is attached below.
[config.log](https://github.com/underworldcode/underworld2/files/3051187/config.log)

**Comment by tingyang2004** [comment]
Any progress on it?

**Comment by jmansour** [comment]
Hi Ting

I was still waiting on further info from you via email.

**Comment by tingyang2004** [comment]
Hmmm, I sent it to your Monash email this Monday. Maybe some information is sensitive. Could you check if it is put in the trash or spam folder? Thanks.

**Comment by jmansour** [comment]
Ok, was indeed in spam.  I'll have a look tomorrow.

**Comment by tingyang2004** [comment]
John has nicely resolved the issue. I will close the ticket now.
